libpod/container_internal_linux.go

```diff
@@ -1399,3 +1411,42 @@ func (c *Container) getOCICgroupPath() (string, error) {
 		return "", errors.Wrapf(define.ErrInvalidArg, "invalid cgroup manager %s requested", c.runtime.config.CgroupManager)
 	}
 }
+
+func (c *Container) setupOverlayMounts() error {
+	g := generate.Generator{Config: c.config.Spec}
+	// Check if the spec file mounts contain an Overlay mount
+	// If they do, mount the overlay.
+	containerDir, err := c.runtime.store.ContainerDirectory(c.ID())
+	if err != nil {
+		return err
+	}
+
+	for i := range c.config.Spec.Mounts {
+		m := &g.Config.Mounts[i]
+		for _, o := range m.Options {
+			if o == "O" {
+				contentDir, err := overlay.TempDir(containerDir, c.RootUID(), c.RootGID())
+				if err != nil {
+					return errors.Wrapf(err, "failed to create TempDir in the %s directory", containerDir)
+				}
+
+				overlayMount, err := overlay.Mount(contentDir, m.Source, m.Destination, c.RootUID(), c.RootGID(), c.runtime.store.GraphOptions())
+				if err != nil {
+					return errors.Wrapf(err, "creating ovelray failed %q", m.Source)
+				}
+				g.Config.Mounts[i] = overlayMount
+				c.config.Mounts = append(c.config.Mounts, m.Source)
```

Comment:
This is NOT safe. `c.config` is NEVER written to the DB after container creation. We cannot use it for keeping track of mutable state.

We're going to need something to track this in `c.state` - I would suggest something like the existing `BindMounts` map.